iati/core/tests/test_rulesets.py

```diff
@@ -162,57 +162,67 @@ def test_rule_init_no_parameters(self, rule_constructor):
-    def xpath_base(self):
-        """Return valid xpath_base."""
+    def valid_single_context(self):
+        """Return valid context with a single match."""
         return '//root_element'
 
     @pytest.fixture
-    def empty_xpath_base(self):
-        """Empty_xpath_base."""
-        return ''
+    def valid_multiple_context(self):
+        """Return a valid context with multiple matches."""
+        return '//nest'
 
-    @pytest.fixture
-    def rule_basic_init(self, rule_constructor, instantiating_case, xpath_base):
-        """Rule subclass."""
-        return rule_constructor(xpath_base, instantiating_case)
+    @pytest.fixture(params=[
+        'count(condition)>0',
+        'condition'
+    ])
+    def valid_condition_case(self, validating_case, request):
+        """Return a case with optional condition attribute."""
+        condition_validating_case = deepcopy(validating_case)
+        condition_validating_case['condition'] = request.param
+        return condition_validating_case
 
-    # @pytest.fixture
-    # def rule_is_valid_for_case(self, rule_constructor, xpath_base, case_for_is_valid_for):
-    #     """Rule with specific cases for checking the `is_valid_for` function."""
-    #     return rule_constructor(xpath_base, case_for_is_valid_for)
+    @pytest.fixture(params=[
+        'count(condition)>0',
+        'condition'
+    ])
+    def invalid_condition_case(self, invalidating_case, request):
+        """Return a case with optional condition attribute."""
+        condition_invalidating_case = deepcopy(invalidating_case)
+        condition_invalidating_case['condition'] = request.param
+        return condition_invalidating_case
 
     @pytest.fixture
-    def rule_empty_xpath_base(self, rule_constructor, instantiating_case, empty_xpath_base):
-        """Rule subclass with an empty xpath_base."""
-        return rule_constructor(empty_xpath_base, instantiating_case)
+    def rule_basic_init(self, rule_constructor, instantiating_case, valid_single_context):
+        """Rule subclass."""
+        return rule_constructor(valid_single_context, instantiating_case)
 
     @pytest.fixture
-    def rule_valid_empty_case(self, empty_path_case, xpath_base, rule_constructor):
-        """Rule with a valid empty path case."""
-        return rule_constructor(xpath_base, empty_path_case)
+    def rule_is_valid_for(self, rule_constructor, validating_case, valid_single_context):
+        """Rule subclass for validation."""
+        return rule_constructor(valid_single_context, validating_case)
 
     @pytest.fixture
-    def rule_empty_xpath_base_and_paths(self, rule_constructor, empty_xpath_base, empty_path_case):
-        """Rule subclass with an empty xpath_base and empty path case."""
-        return rule_constructor(empty_xpath_base, empty_path_case)
+    def rule_is_invalid_for(self, rule_constructor, invalidating_case, valid_single_context):
+        """Rule with specific cases for checking the `is_valid_for` function."""
```

Comment:
Docstring could be improved to clarify that the Rule will be invalid against the relevant test dataset.